torchgeo/trainers/landcoverai.py

```diff
@@ -164,55 +201,53 @@ def validation_step(  # type: ignore[override]
             plt.close()
 
     def validation_epoch_end(self, outputs: Any) -> None:
-        """Logs epoch level validation metrics."""
-        self.log("val_acc", self.val_accuracy.compute())
-        self.log("val_iou", self.val_iou.compute())
-        self.val_accuracy.reset()
-        self.val_iou.reset()
+        """Logs epoch level validation metrics.
+
+        Args:
+            outputs: list of items returned by validation_step
+        """
+        self.log_dict(self.val_metrics.compute())
+        self.val_metrics.reset()
 
     def test_step(  # type: ignore[override]
         self, batch: Dict[str, Any], batch_idx: int
     ) -> None:
-        """Test step identical to the validation step."""
+        """Test step identical to the validation step.
+
+        Args:
+            batch: Current batch
+            batch_idx: Index of current batch
+        """
         x = batch["image"]
-        y = batch["mask"]
+        y = batch["mask"].long().squeeze()
         y_hat = self.forward(x)
         y_hat_hard = y_hat.argmax(dim=1)
 
         loss = self.loss(y_hat, y)
 
         # by default, the test and validation steps only log per *epoch*
-        self.log("test_loss", loss)
-        self.test_accuracy(y_hat_hard, y)
-        self.test_iou(y_hat_hard, y)
+        self.log("test_loss", loss, on_step=False, on_epoch=True)
+        self.test_metrics(y_hat_hard, y)
 
     def test_epoch_end(self, outputs: Any) -> None:
-        """Logs epoch level test metrics."""
-        self.log("test_acc", self.test_accuracy.compute())
-        self.log("test_iou", self.test_iou.compute())
-        self.test_accuracy.reset()
-        self.test_iou.reset()
+        """Logs epoch level test metrics.
+
+        Args:
+            outputs: list of items returned by test_step
+        """
+        self.log_dict(self.test_metrics.compute())
+        self.test_metrics.reset()
 
     def configure_optimizers(self) -> Dict[str, Any]:
-        """Initialize the optimizer and learning rate scheduler."""
-        optimizer: torch.optim.optimizer.Optimizer
-        if self.hparams["optimizer"] == "adamw":
-            optimizer = torch.optim.AdamW(
-                self.model.parameters(),
-                lr=self.hparams["learning_rate"],
-            )
-        elif self.hparams["optimizer"] == "sgd":
-            optimizer = torch.optim.SGD(
-                self.model.parameters(),
-                lr=self.hparams["learning_rate"],
-                momentum=0.9,
-                weight_decay=1e-2,
-            )
-        else:
-            raise ValueError(
-                f"Optimizer choice '{self.hparams['optimizer']}' is not valid."
-            )
+        """Initialize the optimizer and learning rate scheduler.
 
+        Returns:
+            a "lr dict" according to the pytorch lightning documentation
```

Comment:
Can you elaborate on this a bit more?

Reply:
This could be a lot of different things, see https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers.

Should we just link to the docs in that docstring?

Reply:
Yeah, a link would be good

Reply:
Done for landcoverai, so2sat, and chesapeake